moviebot/controller/controller_flask.py

```diff
@@ -0,0 +1,398 @@
+                "rating": row[4],
+                "duration": row[6],
+                "summary": row[10],
+                "image_url": row[9],
+                "imdb_link": row[12],
+            }
+
+    def execute_agent(self, configuration: Dict[str, Any]) -> None:
+        """Configures the controller.
+
+        Args:
+            configuration: Configuration for the agent.
+        """
+        self.configuration = configuration
+        self.configuration["new_user"] = {}
+        if self.configuration["BOT_HISTORY"]["path"]:
+            self.path = self.configuration["BOT_HISTORY"]["path"]
+
+    def restart(self, user_id: str) -> Callable:
+        """Restarts agent for this user."""
+        return self.start_agent(user_id, True)
+
+    def start_agent(self, user_id: str, restart=False) -> HTTP_OBJECT_MESSAGE:
+        """Starts conversation with agent.
+
+        Args:
+            user_id: User id.
+            restart: Whether or not to restart the agent.
+
+        Returns:
+            Object with start message to send to the server.
+        """
+        self.agent[user_id] = Agent(self.configuration)
+
+        (
+            self.agent_response[user_id],
+            self.record_data_agent[user_id],
+            self.user_options[user_id],
+        ) = self.agent[user_id].start_dialogue()
+        if restart:
+            (
+                self.agent_response[user_id],
+                self.user_options[user_id],
+                self.record_data_agent[user_id],
+            ) = self.agent[user_id].start_dialogue(None, restart)
+            return self.user_messages[user_id].text(
+                self.agent_response[user_id], intent=self.agent_intent
+            )
+        else:
+            return {"message": {"text": "", "intent": self.agent_intent}}
+
+    def get_movie_id(self, response: str) -> str:
+        """Retrieves movie id from agent response string.
+
+        Args:
+            response: Agent response.
+
+        Returns:
+            Movie id.
+        """
+        if "/tt" in response:
+            start = response.find("/tt")
+            movie_id = response[start + 3 : start + 10]
+            return movie_id
+
+    def continue_dialogue(self, user_id: str, payload: str) -> None:
+        """Continues dialogue with the agent. Updates movie info.
+
+        Args:
+            user_id: User id.
+            payload: User payload.
+        """
+        user_utterance = UserUtterance({"text": payload})
+        (
+            self.agent_response[user_id],
+            self.user_options[user_id],
+            self.record_data_agent[user_id],
+        ) = self.agent[user_id].continue_dialogue(
+            user_utterance, self.user_options[user_id]
+        )
+        if self.users[user_id]:
+            self.record(user_id, payload)
+        movie_id = self.get_movie_id(self.agent_response[user_id])
+        self.movie_info(movie_id, user_id)
+
+    def store_user(self, user_id: str) -> Callable:
```

Comment:
Ambiguous name. Is this method storing user?

Reply:
It is not, this is related to the storage policy. The method was renamed.